File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/example/thymeleafjava/web/layout/LayoutViewComponent.java
package de.tschuehly.example.thymeleafjava.web.layout;

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext;

@ViewComponent
public
class LayoutViewComponent {

    private record LayoutView(ViewContext nestedViewComponent) implements ViewContext {
    }

    public ViewContext render(ViewContext nestedViewComponent) {
        return new LayoutView(nestedViewComponent);
    }
}

File: thymeleaf/src/test/kotlin/de/tschuehly/spring/viewcomponent/thymeleaf/application/web/layout/LayoutViewComponent.kt
package de.tschuehly.spring.viewcomponent.thymeleaf.application.web.layout

import de.tschuehly.spring.viewcomponent.core.ViewComponent
import de.tschuehly.spring.viewcomponent.core.toProperty
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext

@ViewComponent
class LayoutViewComponent {
    fun render(nestedViewComponent: ViewContext) = ViewContext(
        "nestedViewComponent" toProperty nestedViewComponent
    )
}